### ISSUE TITLE
Documentation: literal "types"

### ISSUE THREAD

**Issue by bodawei** [microsoft/TypeScript#18102]
**Code**

Try the following out at https://www.typescriptlang.org/play/index.html  (sadly, this is too long to include a sharable link)
```ts
let num1: 1;
let num2: 2;
let num3: typeof num2;
num3 = num2;
num3 = num1;
```

**Expected behavior:**
This works exactly as expected, such that there's an error on the second assignment to num3.  

I don't easily see something in your doc that mentions this is possible.  It is probably implicit in various places, as it is particularly useful with some unions. Seems like calling this out explicitly would be valuable?

**Comment by mhegazy** [comment]
The spec needs to be updated for these. 

Some relevant readings meanwhile, literal types are widened to their base type in mutable locations (see https://github.com/Microsoft/TypeScript/pull/10676), and explicit literal type annotations are not widened (see https://github.com/Microsoft/TypeScript/pull/11126)

**Comment by DanielRosenwasser** [comment]
Another relevant change is that our section on specialized signatures needs to be simplified. As far as I remember, specialized signatures are only re-sorted now.

**Comment by RyanCavanaugh** [comment]
The spec is now an "archived" artifact and we won't be tracking defects against it anymore.
